packages/dart_firebase_admin/lib/src/auth/token_verifier.dart

```diff
@@ -249,6 +247,17 @@ class TokenProvider {
     required this.tenant,
   });
 
+  @internal
+  factory TokenProvider.fromMap(Map<dynamic, dynamic> map) {
+    return TokenProvider(
+      identities: map['identities']! as Map<String, Object?>,
+      signInProvider: map['sign_in_provider']! as String,
```

Comment:
In order for this PR to solve #56, `TokenProvider.signInProvider` should be made optional, and here we would need to do the following:
```suggestion
      signInProvider: map['sign_in_provider'] as String?,
```

cc @labrom